system/topic_state_monitor/Readme.md

```diff
@@ -35,23 +35,25 @@ The types of topic status and corresponding diagnostic status are following.
+| `transient_local` | bool   | false         | QoS policy of topic subscription (Transient Local/Volatile)   |
+| `best_effort`     | bool   | false         | QoS policy of topic subscription (Best Effort/Reliable)       |
+| `diag_name`       | string | -             | Name used for the diagnostics to publish                      |
+| `update_rate`     | double | 10.0          | Timer callback period [Hz]                                    |
 
 ### Core Parameters
 
-| Name              | Type   | Default Value | Description                                                                                          |
-| ----------------- | ------ | ------------- | ---------------------------------------------------------------------------------------------------- |
-| `topic`           | string | -             | Name of target topic                                                                                 |
-| `topic_type`      | string | -             | Type of target topic                                                                                 |
-| `transient_local` | bool   | false         | QoS policy of topic subscription (Transient Local/Volatile)                                          |
-| `best_effort`     | bool   | false         | QoS policy of topic subscription (Best Effort/Reliable)                                              |
-| `diag_name`       | string | -             | Name used for the diagnostics to publish                                                             |
-| `warn_rate`       | double | 0.5           | If the topic rate is lower than this value, the topic status becomes `WarnRate`                      |
-| `error_rate`      | double | 0.1           | If the topic rate is lower than this value, the topic status becomes `ErrorRate`                     |
-| `timeout`         | double | 1.0           | If the topic subscription is stopped for more than this time [s], the topic status becomes `Timeout` |
+| Name          | Type   | Default Value | Description                                                                                          |
+| ------------- | ------ | ------------- | ---------------------------------------------------------------------------------------------------- |
+| `warn_rate`   | double | 0.5           | If the topic rate is lower than this value, the topic status becomes `WarnRate`                      |
+| `error_rate`  | double | 0.1           | If the topic rate is lower than this value, the topic status becomes `ErrorRate`                     |
+| `timeout`     | double | 1.0           | If the topic subscription is stopped for more than this time [s], the topic status becomes `Timeout` |
+| `window_size` | int    | 10            | Window size of target topic for calculating frequency                                                |
 
 ## Assumptions / Known limits
 
```

Comment:
What made you change the categories of parameters?
I feel it should be done in another PR to minimize the diff of this PR.

Reply:
I have split the PR.
https://github.com/autowarefoundation/autoware.universe/pull/1586

Reply:
Thank you very much! :pray: